src/mapreduce.jl

```diff
@@ -90,6 +90,36 @@ end
     end
 end
 
+@generated function reduce{D}(op, a::StaticArray, ::Type{Val{D}})
+    S = size(a)
+    if S[D] == 1
+        return :(return a)
+    else
+        N = ndims(a)
+        Snew = ([n==D ? 1:S[n] for n = 1:N]...)
```

Comment:
The spacing around the ternary here is pretty confusing, as `1:S[n]` ends up looking like a range.  I'd space that out a bit.